File: .tmp_py4013/batch_12_dump/3344_maximum_sized_array.js
// LeetCode 3344 - Maximum Sized Array
// https://leetcode.com/problems/maximum-sized-array/

function ok(n, s) {
    let sum = 0;
    for (let i = 0; i < n; i++) {
        for (let j = 0; j < n; j++) {
            const ij = i | j;
            sum += ij * (n - 1) * n / 2;
            if (sum > s) return false;
        }
    }
    return sum <= s;
}
var maxSizedArray = function(s) {
    let lo = 1, hi = 2000;
    while (lo < hi) {
        const mid = Math.floor((lo + hi + 1) / 2);
        if (ok(mid, s)) lo = mid;
        else hi = mid - 1;
    }
    return lo;
};

File: .tmp_rs4013/rust_check16/3344_maximum_sized_array.rs
struct Solution;
// LeetCode 3344 - Maximum Sized Array
// https://leetcode.com/problems/maximum-sized-array/

impl Solution {
    pub fn max_sized_array(s: i64) -> i32 {
        let ok = |n: i64| -> bool {
            let mut sum = 0i64;
            for i in 0..n {
                for j in 0..n {
                    let ij = i | j;
                    sum += ij * (n - 1) * n / 2;
                    if sum > s {
                        return false;
                    }
                }
            }
            sum <= s
        };
        let mut lo = 1i64;
        let mut hi = 2000i64;
        while lo < hi {
            let mid = (lo + hi + 1) / 2;
            if ok(mid) {
                lo = mid;
            } else {
                hi = mid - 1;
            }
        }
        lo as i32
    }
}

fn main() {}
